packages/key-management/src/util/mapHardwareSigningData.ts

```diff
@@ -395,15 +401,12 @@ const prepareTrezorMintBundle = (
         for (let k = 0; k < assetsKeys.len(); k++) {
           const assetName = scope.manage(assetsKeys.get(k));
           const amount = scope.manage(assets.get(assetName));
-          const positiveAmount = scope.manage(amount?.as_positive())?.to_str();
-          const negativeAmount = scope.manage(amount?.as_negative())?.to_str();
-          if (!amount || !positiveAmount || !negativeAmount) {
-            throw new HwMappingError('Missing token amount.');
+          if (amount) {
+            tokens.push({
```

Comment:
I'm not sure if it's correct, but the logic has changed a bit.
Previously, in case of `undefined` amount it would throw.
Now, in case of `undefined` amount it will just not add the token.

Reply:
No longer exists in 67e3f4aa61259851b735f3652c1fd95e64000848

Reply:
The original code threw a `HwMappingError('Missing token amount.')` in case of falsy `amount`.
The current code doesn't throw anymore. It simply skips adding it to the token map.
If that's the desired flow, I'm ok with it, and we can mark this resolved 🤝

Reply:
That logic was only there due to the design of the CML needing multiple steps to get the value, so this is more of a type issue. I've improved this in c2cec975920ae58f76deb6ee5c084722bd54a2d6